src/plugins/data/public/ui/query_editor/query_editor_top_row.tsx

```diff
@@ -90,26 +90,51 @@ export default function QueryEditorTopRow(props: QueryEditorTopRowProps) {
     }
   }, []);
 
+  // Memoized callback for focusing query bar
+  const handleFocusQueryBar = useCallback(() => {
+    // Monaco-based query editor
+    const queryEditorInput = document.querySelector(DATA_DOM_SELECTORS.QUERY_EDITOR_INPUT);
+    if (queryEditorInput) {
+      // Look for Monaco editor within the query editor container
+      const monacoEditor = queryEditorInput.querySelector(
+        '.monaco-editor textarea'
+      ) as HTMLTextAreaElement;
+      if (monacoEditor) {
+        monacoEditor.focus();
+        return;
+      }
+    }
+  }, []);
+
   keyboardShortcut?.useKeyboardShortcut({
     id: 'open_date_picker',
     pluginId: 'data',
     name: 'Open Date Picker',
-    category: 'Open',
-    keys: 'shift+t',
+    category: 'Search',
+    keys: 'shift+d',
     execute: handleOpenDatePicker,
   });
 
   keyboardShortcut?.useKeyboardShortcut({
     id: 'refresh_query',
     pluginId: 'data',
     name: 'Refresh Results',
-    category: 'query',
-    keys: 'shift+r',
+    category: 'Data actions',
+    keys: 'r',
     execute: () => {
       onClickSubmitButton({ preventDefault: () => {} } as React.MouseEvent<HTMLButtonElement>);
     },
   });
 
+  keyboardShortcut?.useKeyboardShortcut({
+    id: 'focus_query_bar',
+    pluginId: 'data',
+    name: 'Focus Query Bar',
```

Comment:
we are kind of mixing up terms here. this would be the query editor.  the query bar is something within dashboards.